lib/new_relic/telemetry/plug.ex

```diff
@@ -85,18 +91,20 @@ defmodule NewRelic.Telemetry.Plug do
 
   @doc false
   def handle_event(
-        @cowboy_start,
-        %{system_time: system_time},
+        [server, :request, :start],
+        measurements,
         meta,
         _config
       ) do
     Transaction.Reporter.start_transaction(:web)
 
+    system_time = get_system_time(measurements)
+
     if NewRelic.Config.enabled?(),
-      do: DistributedTrace.start(:http, meta.req.headers)
+      do: DistributedTrace.start(:http, get_headers(meta, server))
```

Comment:
We could avoid converting headers to a map twice by gathering the headers into a variable before this line and passing them along to this function and `maybe_report_queueing`